cookbook/activity_manager/add_project_remover.rst

```diff
@@ -0,0 +1,150 @@
+===================================================
+
+.. _reference: ../../reference/activity_manager/catalog_impact.html
+
+Several catalog updates can impact projects integrity. In this case we remove impacted projects. Akeneo PIM already
+manages its own catalog updates. The list is available in Activity Manager reference_ part.
+
+How does It Work?
+_________________
+
+A subscriber handles catalog structure updates listening to `Akeneo\Component\StorageUtils\StorageEvents::POST_SAVE`
+and `Akeneo\Component\StorageUtils\StorageEvents::PRE_REMOVE` on all entities and dispatching them to the
+`PimEnterprise\Component\ActivityManager\Remover\ChainedProjectRemover`. This chained project remover contains all
+project removers and guesses which one has the responsibility to handle both entity and action it received.
+
+A `ProjectRemover` has the responsibility to find and remove all projects impacted by entity and action pair.
+
+How To Add A Custom Project Remover?
+____________________________________
+
+Reference data will be a good example to demonstrate how to add a `ProjectRemover`. We will take the example of Color
+reference data.
+
+Each project remover must implement the `PimEnterprise\Component\ActivityManager\Remover\ProjectRemoverInterface` and
+the service must be tagged as `pimee_activity_manager.project_remover`.
+
+.. code-block:: php
+
+    // src/AcmeEnterprise/Bundle/AppBundle/Remover/ColorProjectRemover.php
+    <?php
+
+    namespace AcmeEnterprise\Bundle\AppBundle\Remover;
+
+    use AcmeEnterprise\Bundle\AppBundle\Entity\Color;
+    use Akeneo\Component\StorageUtils\Detacher\ObjectDetacherInterface;
+    use Akeneo\Component\StorageUtils\Remover\RemoverInterface;
+    use Akeneo\Component\StorageUtils\StorageEvents;
+    use Doctrine\Common\Persistence\ObjectRepository;
+    use PimEnterprise\Component\ActivityManager\Model\ProjectInterface;
+    use PimEnterprise\Component\ActivityManager\Remover\ProjectRemoverInterface;
+
+    class ColorProjectRemover implements ProjectRemoverInterface
+    {
+        /** @var RemoverInterface */
+        protected $projectRemover;
+
+        /** @var ObjectRepository */
+        protected $projectRepository;
+
+        /** @var ObjectDetacherInterface */
+        protected $detacher;
+
+        /**
+         * @param ObjectRepository        $projectRepository
+         * @param RemoverInterface        $projectRemover
+         * @param ObjectDetacherInterface $detacher
+         */
+        public function __construct(
+            ObjectRepository $projectRepository,
+            RemoverInterface $projectRemover,
+            ObjectDetacherInterface $detacher
+        ) {
+            $this->projectRepository = $projectRepository;
+            $this->projectRemover = $projectRemover;
+            $this->detacher = $detacher;
+        }
+
+        /**
+         * `isSupported` method is called by the ChainedProjectRemover to know which ProjectRemover fit entity/action
```

Comment:
This method [...] to determinate which `ProjectRemover` matches the entity/action pair it receives.